### PR TITLE
Add evaluateRulesets action, subscribeRulesetItems event for "in browser messages" beta

### PR DESCRIPTION
**Pull request by jonsnyder**
## Description

This adds evaluateRulesets action, and subscribeRulesetItems event for the In browser messages beta work. It also adds a field in sendEvent for "decisionContext".

I still need to add functional tests, and add beta labels.

## Screenshots (if appropriate):

Screenshot 2023-10-23 at 11 33 29 AM
Screenshot 2023-10-23 at 11 31 47 AM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.
- [ ] I've updated the schema in extension.json or no changes are necessary.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Comment by jasonwaters** [comment]
this PR will need to be modified to support moving `decisionContext` within `personalization` if that pr is merged https://github.com/adobe/alloy/pull/1068

**Comment by jfkhoury** [comment]
Looks good, @jonsnyder, thank you for making the edits.

**Review by jfkhoury — APPROVED** [review]

File: src/view/forms/simpleMap.jsx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+            path: `${context.path}.key`,
+            message: `Please provide a ${keyLabel.toLowerCase()}.`
+          });
+        }
+
+        const { path, parent } = context;
+        const items = [...parent];
+        const currentIndex = items.indexOf(item);
+        const previousItems = items.slice(0, currentIndex);
+        console.log("unique key validator 2", currentIndex, previousItems);
```
Comment (carterworks) [inline src/view/forms/simpleMap.jsx:93]: TODO (eventually): remove this console log

File: src/view/components/formikReactSpectrum3/formikComboBox.jsx [hunk 2]
```diff
@@ -29,6 +29,7 @@ const FormikComboBox = ({ name, width, validate, ...otherProps }) => {
     name,
     validate
   });
+  console.log("FormikComboBox", otherProps);
```
Comment (carterworks) [inline src/view/components/formikReactSpectrum3/formikComboBox.jsx:32]: TODO (eventually): remove this console log

**Review by carterworks — APPROVED** [review]

File: src/view/actions/evaluateRulesets.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+const evaluateRulesetsForm = form(
+  {
+    wrapGetInitialValues,
+    wrapGetSettings
+  },
+  [
+    notice({
+      title: "Evaluate rulesets action",
+      description:
+        "This action manually triggers ruleset evaluation. Rulesets are returned from Adobe Journey Optimizer in-browser-messages.",
```
Comment (jasonwaters) [inline src/view/actions/evaluateRulesets.jsx:47]: I would revise this just a tad:

> This action manually triggers ruleset evaluation. Rulesets are returned by Adobe Journey Optimizer to support features like in-browser messages.

**Review by jasonwaters — APPROVED** [review]

**Pull request merged**